CMSgov/mint-app
# Easi 2256/follow model plan

# EASI-2256

## Changes and Description

- Add Plan_Favorite table
     - GQL queries and mutations to access the favorite information directly
- Update Model Plan Object, GQL and SQL queries to return isFavorite whenever you get a model plan directly
    - Earlier implementation required a separate SQL call to check, vs checking at time of access

Note:
 - If the approach of accessing directly from the model is agreed upon, I think we can remove the planFavoriteCollection query from GQL, and have it only be accessible from the model
 - If desired we could look at updating the favorite table as part of the model plan update mutation, but having separate endpoints seems cleaner.

## How to test this change

1. `scripts/dev up:backend:debug && scripts/dev db:seed`
2. Create a plan Favorite
```gql
mutation addPlanFavorite 
{
    addPlanFavorite(modelPlanID:"6e224030-09d5-46f7-ad04-4bb851b36eab"){
        modelPlanID
        userID
    }
}
```

3. Get ModelPlan and see that it shows it is favorited
```gql
query ModelPlan{
  
  modelPlan(id:"6e224030-09d5-46f7-ad04-4bb851b36eab"){
    id
    modelName
    isFavorite
  }
  
  
}
```
4. Delete the plan Favorite Record
```gql
mutation DeletePlanFavorite 
{
    deletePlanFavorite(modelPlanID:"6e224030-09d5-46f7-ad04-4bb851b36eab"){
        modelPlanID
        userID
    }
}
```
## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
